vendor/bat-native-ads/src/bat/ads/internal/base/strings/string_html_parse_util.h

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_BASE_STRINGS_STRING_HTML_PARSE_UTIL_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_BASE_STRINGS_STRING_HTML_PARSE_UTIL_H_
+
+#include <string>
+
+namespace ads {
+
+std::string FindFirstRegexMatch(const std::string& search_text,
+                                const std::string& rgx_str);
```

Comment:
This function does not need to be public as it is only used by ParseTagAttribute. The function can also be a non-member function in an empty namespace in the .cc file

Reply:
re my comment on unit-tests: if you pull that into the .cc as non-member you don't need to add an extra unit test. It should be covered by the `ParseTagAttribute` test

Reply:
This function no longer exists, as a result of updating my process to use RE2